### PR TITLE
Display ip address of worker in `rq info`

### LINKED ISSUE #1386: Possible to get worker IP?
rq info shows the remote worker by their (self-declared) hostname. Is it possible to get their IPs as well?

### PR DESCRIPTION
**Pull request by rpkak**
Now
```
$ rq info
```
returns:
```
0 queues, 0 jobs total

0a9259efacac4bfda0eac0694b04a127 (9ae8ebd36f1e 172.17.0.3:57666 1): idle default
1 workers, 0 queues

Updated: 2021-05-07 15:45:52.50358
```

(I'm using docker for this network.)

fix #1386

### CODE REVIEW

**Comment by selwin** [comment]
@rpkak can you please update the PR so that the tests pass?

**Comment by selwin** [comment]
It would be great if we can get this PR ready this weekend, I'll make a release by Monday.

**Comment by rpkak** [comment]
> @rpkak can you please update the PR so that the tests pass?

fixed

File: rq/worker.py [hunk 1]
```diff
@@ -225,6 +218,16 @@ def __init__(self, queues, name=None, default_result_ttl=DEFAULT_RESULT_TTL,
 
         self.disable_default_exception_handler = disable_default_exception_handler
 
+        if prepare_for_work:
+            self.hostname = socket.gethostname()
+            self.pid = os.getpid()
+            connection.client_setname(self.name)
```
Comment (selwin) [inline rq/worker.py:224]: What does this actually do? I'm not sure why we're making changes to Redis connection.

Comment (rpkak) [reply rq/worker.py:224]: It sets the client name of the redis connection. It is used for getting the current connection to get the ip from.

Try `CLIENT LIST` in the redis cli.

File: rq/worker.py [hunk 2]
```diff
@@ -225,6 +218,16 @@ def __init__(self, queues, name=None, default_result_ttl=DEFAULT_RESULT_TTL,
...
+            self.address = [client['addr'] for client in connection.client_list() if client['name'] == self.name][0]
```
Comment (selwin) [inline rq/worker.py:225]: Can we change `address` to `ip_address`?

File: rq/worker.py [hunk 3]
```diff
@@ -301,6 +304,7 @@ def register_birth(self):
                 'queues': queues,
                 'pid': self.pid,
                 'hostname': self.hostname,
+                'address': self.address,
```
Comment (selwin) [inline rq/worker.py:307]: Please change this to `ip_address`

File: rq/worker.py (hunk 2 above)
Comment (rpkak) [reply rq/worker.py]: done

File: rq/worker.py (hunk 3 above)
Comment (rpkak) [reply rq/worker.py]: done

**Comment by selwin** [comment]
Thanks!

**Pull request merged**